experimental/printer/dom/chunks_test.go

```diff
@@ -0,0 +1,294 @@
+		inner1.Insert(newChunk("(", SplitKindNever, false, 2))
+		inner1.Insert(newChunk("nested_nested_e ", SplitKindNever, true, 2))
+		inner1.Insert(newChunk("== ", SplitKindNever, true, 2))
+		inner1.Insert(newChunk("nested_nested_f", SplitKindNever, false, 2))
+		closeParen2 := newChunk(") ", SplitKindSoft, true, 2)
+
+		braces2 := NewChunks()
+		openBrace2 := newChunk("{ ", SplitKindSoft, true, 2)
+
+		inner2 := NewChunks()
+		inner2.Insert(newChunk("nested_nested_method_b(", SplitKindNever, false, 3))
+		inner2.Insert(newChunk("); ", SplitKindSoft, true, 3))
+		inner2.Insert(newChunk("while ", SplitKindNever, true, 3))
+		inner2.Insert(newChunk("(", SplitKindNever, false, 3))
+		inner2.Insert(newChunk("nested_nested_nested_g ", SplitKindNever, true, 3))
+		inner2.Insert(newChunk("== ", SplitKindNever, true, 3))
+		inner2.Insert(newChunk("     ", SplitKindUnknown, false, 3))
+		inner2.Insert(newChunk("nested_nested_nested_h ", SplitKindNever, false, 3))
+		closeParen3 := newChunk(") ", SplitKindSoft, true, 3)
+
+		braces3 := NewChunks()
+		openBrace3 := newChunk("{ ", SplitKindSoft, true, 3)
+
+		inner3 := NewChunks()
+		inner3.Insert(newChunk("        ", SplitKindUnknown, false, 4))
+		inner3.Insert(newChunk("nested_nested_nested_method_c(", SplitKindNever, false, 4))
+		inner3.Insert(newChunk("); ", SplitKindSoft, true, 4))
+
+		openBrace3.SetChild(inner3)
+		braces3.Insert(openBrace3)
+		braces3.Insert(newChunk("} ", SplitKindSoft, true, 3))
+
+		closeParen3.SetChild(braces3)
+		inner2.Insert(closeParen3)
+		inner2.Insert(newChunk("nested_nested_method_d(", SplitKindNever, false, 3))
+		inner2.Insert(newChunk("); ", SplitKindSoft, true, 3))
+		inner2.Insert(newChunk("      ", SplitKindUnknown, false, 3))
+		inner2.Insert(newChunk("nested_nested_method_e(", SplitKindNever, false, 3))
+		inner2.Insert(newChunk("); ", SplitKindSoft, true, 3))
+
+		openBrace2.SetChild(inner2)
+		braces2.Insert(openBrace2)
+		braces2.Insert(newChunk("} ", SplitKindSoft, true, 2))
+
+		closeParen2.SetChild(braces2)
+		inner1.Insert(closeParen2)
+		inner1.Insert(newChunk("nested_method_f(", SplitKindNever, false, 2))
+		inner1.Insert(newChunk("   ", SplitKindUnknown, false, 2))
+		inner1.Insert(newChunk("); ", SplitKindSoft, true, 2))
+
+		openBrace1.SetChild(inner1)
+		braces1.Insert(openBrace1)
+		braces1.Insert(newChunk("} ", SplitKindSoft, true, 1))
+
+		closeParen1.SetChild(braces1)
+		inner0.Insert(closeParen1)
+		inner0.Insert(newChunk("method_g(", SplitKindNever, false, 1))
+		inner0.Insert(newChunk("); ", SplitKindSoft, true, 1))
+
+		openBrace0.SetChild(inner0)
+		braces0.Insert(openBrace0)
+		braces0.Insert(newChunk("}", SplitKindSoft, false, 0))
+
+		closeParen0.SetChild(braces0)
+		block.Insert(closeParen0)
+
+		require.Equal(t, unformatted, block.output())
+
+		// Set math.MaxInt32 to line limit (basically ridiculously long/effectively no character limit).
+		// Set 2-space indent string.
+		block.format(math.MaxInt32, "  ")
```

Comment:
I think I should add a test case for formatting behaviour with the line limit restrictions.